src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Validators/SortStringValidators.cs

```diff
@@ -0,0 +1,92 @@
+using System.Text.RegularExpressions;
+using FluentValidation;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Validators.ErrorDetails;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Model;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Validators;
+
+public static partial class SortStringValidators
+{
+    private const string ExpectedFormat = "{field}|{order}";
+    private const int MaxFieldLength = 40;
+
+    public static IRuleBuilderOptionsConditions<T, string> SortString<T>(this IRuleBuilder<T, string> rule)
+    {
+        return rule.Custom((value, context) =>
+        {
+            if (!HasValidFormat(value))
+            {
+                context.AddFailure(
+                    message: ValidationMessages.SortFormat,
+                    detail: new FormatErrorDetail(value, ExpectedFormat: ExpectedFormat)
+                );
+
+                return;
+            }
+
+            var sort = new ParsedSort(value);
+
+            if (!HasValidFieldLength(sort))
+            {
+                context.AddFailure(
+                    message: ValidationMessages.SortMaxFieldLength,
+                    detail: new MaxFieldLengthErrorDetail(sort.String),
+                    new Dictionary<string, object>
+                    {
+                        { "MaxLength", MaxFieldLength }
+                    }
+                );
+            }
+
+            if (!HasValidDirection(sort))
+            {
+                context.AddFailure(
+                    message: ValidationMessages.SortDirection,
+                    detail: new DirectionErrorDetail(sort.String)
+                );
+            }
+        });
+    }
+
+    [GeneratedRegex(@"^[\w_]+\|[A-Za-z]+$", RegexOptions.Compiled, matchTimeoutMilliseconds: 200)]
```

Comment:
Maybe we should add this `RegexOptions.Compiled` option to the generated regex (`PeriodRegex` method) in `TimePeriodFormatter.cs` in the utils.

Don't have to do as part of this PR though

Reply:
Done. I've also set `matchTimeoutMilliseconds` to 200 too